### PR TITLE
Site Editor: Replace incorrect `find` with `filter` in the sidebar

### PR DESCRIPTION
**Pull request by Copons**
## Description

This PR fixes what appears to be a simple copy-paste typo (compare it to the [templates pages menu](https://github.com/WordPress/gutenberg/blob/09d43d6b0f6bb795bd974079352483af208680bc/packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/templates-pages.js#L22)).

Using `find` on a dynamically fetched list can produce unexpected behaviour (on top of returning the incorrect value), which propagates to the rendered `map`, causing plenty of duplicated keys (e.g. `wp_template-undefined`).

## How has this been tested?

- Open the Site Editor, and open the browser console.
- The sidebar should be mounted and its content loaded behind the scenes, but just to be sure, try opening it anyway.
- Check that there are no "duplicated key" errors reported inside `NavigationGroup`.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by david-szabo97 — APPROVED** [review]
Oops 😅 Don't blame me 😆

**Comment by Addison-Stavlo** [comment]
Pardon my ignorance, Im just trying to understand how this was causing the duplicate problem.

`find` instead of `filter` would return one item instead of an array of items.  So it seems the `map` would only output 1 item instead of the entire list.  That one item referenced from a dynamic list may update and be outdated, so the `wp_template-undefined` makes sense... but how are there duplicates of that key given that there would only be 1 item mapped here and there is only 1 instance of the component? 🤔

**Comment by Copons** [comment]
> Pardon my ignorance, Im just trying to understand what was going on here and how this was causing the duplicate problem.
> 
> `find` instead of `filter` would return one item instead of an array of items. So it seems the `map` would only output 1 item instead of the entire list. That one item referenced from a dynamic list may update and be outdated, so the `wp_template-undefined` makes sense... but how are there duplicates of that key given that there would only be 1 item mapped here and there is only 1 instance of the component? 🤔

@Addison-Stavlo It all comes down to Lodash's `map` which is able to loop through both arrays and objects.

Whereas `filter` always returns an array, `find` returns either undefined or a the first found element, which in this case is a template object.
When we send that template object to Lodash `map`, it starts looping through its **properties** instead of just using it as a 1-element array.
Inside the `map`, we use `template.id` to build the component key, but `id` is a template property, not a "template property's property". 😄 

In other words, this happens:
```js
const templates = [ { id: 1, slug: 'index' } ];

const filtered = templates.filter( ( { slug } ) => slug === 'index' );
map( filtered, item => console.log( item ) );
// { id: 1, slug: 'index' }

const found = templates.find( ( { slug } ) => slug === 'index' );
map( found, item => console.log( item ) );
// 1
// 'index'
```

**Comment by Addison-Stavlo** [comment]
Ohh!  Wow gotcha, that makes sense.  Thanks for explaining. 😁
